app/src/main/java/zegreanu/cristi/geoquiz/QuizActivity.kt

```diff
@@ -3,39 +3,66 @@ package zegreanu.cristi.geoquiz
 import android.os.Bundle
 import android.view.Gravity
 import android.widget.Button
+import android.widget.TextView
 import android.widget.Toast
 import androidx.appcompat.app.AppCompatActivity
 
 class QuizActivity : AppCompatActivity() {
     private lateinit var trueButton: Button
     private lateinit var falseButton: Button
+    private lateinit var nextButton: Button
+    private lateinit var questionTextView: TextView
+
+    private var questionBank = arrayOf(
+        Question(R.string.question_australia, true),
+        Question(R.string.question_oceans, true),
+        Question(R.string.question_mideast, false),
+        Question(R.string.question_africa, false),
+        Question(R.string.question_americas, true),
+        Question(R.string.question_asia, true)
+    )
+    private var currentIndex: Int = 0
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_quiz)
 
+        questionTextView = findViewById(R.id.question_text_view)
+        updateQuestion()
+
         trueButton = findViewById(R.id.true_button)
         trueButton.setOnClickListener {
-            Toast.makeText(
-                this@QuizActivity,
-                R.string.correct_toast,
-                Toast.LENGTH_SHORT
-            ).run {
-                setGravity(Gravity.TOP, 0, 0)
-                show()
-            }
+            checkAnswer(true)
         }
 
         falseButton = findViewById(R.id.false_button)
         falseButton.setOnClickListener {
-            Toast.makeText(
-                this@QuizActivity,
-                R.string.incorrect_toast,
-                Toast.LENGTH_SHORT
-            ).run {
-                setGravity(Gravity.TOP, 0, 0)
-                show()
-            }
+            checkAnswer(false)
         }
+
+        nextButton = findViewById(R.id.next_button)
+        nextButton.setOnClickListener {
+            currentIndex = (currentIndex + 1) % questionBank.size
+            updateQuestion()
+        }
+    }
+
+    private fun updateQuestion() {
+        val question = questionBank[currentIndex].textResId
+        questionTextView.setText(question)
+    }
+
+    private fun checkAnswer(userPressedTrue: Boolean) {
+        val answerIsTrue = questionBank[currentIndex].answerTrue
+        val messageResId = if (userPressedTrue == answerIsTrue)
```

Comment:
You might be able to use a shorthand if statement here

Reply:
afaik, kotlin doesn't have that feature ;)

Reply:
code improved with oneliner "if (userPressedTrue == answerIsTrue) R.string.correct_toast else R.string.incorrect_toast"